optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/io/jaxb/OffsetDateTimeJaxbAdapter.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.benchmark.impl.io.jaxb;
+
+import java.time.DateTimeException;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.ChronoField;
+
+import javax.xml.bind.annotation.adapters.XmlAdapter;
+
+public class OffsetDateTimeJaxbAdapter extends XmlAdapter<String, OffsetDateTime> {
+    private final DateTimeFormatter formatter;
+
+    public OffsetDateTimeJaxbAdapter() {
+        formatter = new DateTimeFormatterBuilder()
+                .appendPattern("uuuu-MM-dd'T'HH:mm:ss")
+                .appendFraction(ChronoField.NANO_OF_SECOND, 0, 9, true)
+                .appendOffsetId()
+                .toFormatter();
+    }
+
+    @Override
+    public OffsetDateTime unmarshal(String offsetDateTimeString) {
+        if (offsetDateTimeString == null || offsetDateTimeString.isEmpty()) {
+            return null;
+        }
```

Comment:
I think that this is a reason to throw.

Reply:
I see. Some fields of the serialized model, though, can be null and in such a case throwing an exception is not an option. Let me see if JAXB can do something about it without just passing null to the adapter.

Reply:
Unfortunately, JAXB keeps passing null values into the adapters despite corresponding fields are marked with `@XmlElement(nillable = true)`.